backtester/config/config.go

```diff
@@ -0,0 +1,131 @@
+package config
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+
+	gctcommon "github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/common/file"
+	"github.com/thrasher-corp/gocryptotrader/log"
+)
+
+// ReadConfigFromFile will take a config from a path
+func ReadConfigFromFile(path string) (*Config, error) {
+	if !file.Exists(path) {
+		return nil, errors.New("file not found")
+	}
+
+	fileData, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	return LoadConfig(fileData)
+}
+
+// LoadConfig unmarshalls byte data into a config struct
+func LoadConfig(data []byte) (resp *Config, err error) {
+	err = json.Unmarshal(data, &resp)
+	return resp, err
+}
+
+// PrintSetting prints relevant settings to the console for easy reading
+func (cfg *Config) PrintSetting() {
+	log.Info(log.BackTester, "-------------------------------------------------------------")
+	log.Info(log.BackTester, "------------------Backtester Settings------------------------")
+	log.Info(log.BackTester, "-------------------------------------------------------------")
+	log.Info(log.BackTester, "------------------Strategy Settings--------------------------")
+	log.Info(log.BackTester, "-------------------------------------------------------------")
+	log.Infof(log.BackTester, "Strategy: %s", cfg.StrategySettings.Name)
+	if len(cfg.StrategySettings.CustomSettings) > 0 {
+		log.Info(log.BackTester, "Custom strategy variables:")
+		for k, v := range cfg.StrategySettings.CustomSettings {
+			log.Infof(log.BackTester, "%s: %v", k, v)
+		}
+	} else {
+		log.Info(log.BackTester, "Custom strategy variables: unset")
+	}
+	log.Infof(log.BackTester, "Simultaneous Signal Processing: %v", cfg.StrategySettings.SimultaneousSignalProcessing)
+	for i := range cfg.CurrencySettings {
+		log.Info(log.BackTester, "-------------------------------------------------------------")
+		currStr := fmt.Sprintf("------------------%v %v-%v Settings---------------------------------------------------------",
+			cfg.CurrencySettings[i].Asset,
+			cfg.CurrencySettings[i].Base,
+			cfg.CurrencySettings[i].Quote)
+		log.Infof(log.BackTester, currStr[:61])
+		log.Info(log.BackTester, "-------------------------------------------------------------")
+		log.Infof(log.BackTester, "Exchange: %v", cfg.CurrencySettings[i].ExchangeName)
+		log.Infof(log.BackTester, "Initial funds: %.2f", cfg.CurrencySettings[i].InitialFunds)
```

Comment:
The decimal places can be set to %.4f for maker/taker fees 😄

Reply:
Sorry, I'm used to btcmarkets :D